runtime/op/meta/ztests/pruner.yaml

```
@@ -0,0 +1,44 @@
+# This test makes sure the type context from metadata processing doesn't 
+# leak into the flowgraph, except for metaqueries, where it should.
+script: |
+  export ZED_LAKE=test
+  zed init -q
+  zed create -q -orderby k:asc tmp
+  zed use -q tmp
+  seq 10 15 | zq '{k:this}' - | zed load -q -
+  seq 8 12 | zq '{k:this}' - | zed load -q -
+  seq 20 25 | zq '{k:this}' - | zed load -q -
+  seq 14 16 | zq '{k:this}' - | zed load -q -
+  zed query "from tmp:objects tap | k > 18" | zq -z "drop id" -
+  echo ===
+  zed query "from tmp:objects tap | k <= 10" | zq -z "drop id" -
+  echo ===
+  zed query "from tmp:objects tap | k >= 15 and k < 20" | zq -z "drop id" -
+  echo ===
+  zed query  "from tmp:objects tap | k <= 9 or k > 24" | zq -z "drop id" -
+  echo ===
+  zed query  'from tmp:objects tap | a[k] == "foo" or k >= 20' | zq -z "drop id" -
+  echo ===
+  zed query  'from tmp:objects tap | a[k] == "foo" and k >= 20' | zq -z "drop id" -
```

Nit: No need for zq here.
```suggestion
  zed query "from tmp:objects tap | k > 18 | drop id"
  echo ===
  zed query "from tmp:objects tap | k <= 10 | drop id"
  echo ===
  zed query "from tmp:objects tap | k >= 15 and k < 20 | drop id"
  echo ===
  zed query "from tmp:objects tap | k <= 9 or k > 24 | drop id"
  echo ===
  zed query 'from tmp:objects tap | a[k] == "foo" or k >= 20 | drop id'
  echo ===
  zed query 'from tmp:objects tap | a[k] == "foo" and k >= 20 | drop id'
```

---

While this happens to work, the idea of the "tap" option is to analyze the flowgraph provided then emit the objects in accordance with the downstream flowgraph so tacking on additional processing that would interfere with the analysis being tested seems like a bad idea.  In this case it's ok because the drop op doesn't change the analysis but it might down the road so I'm inclined to leave the zq commands here to emphasize this point.

---

👍